src/containers/AccountConnection.jsx

```diff
@@ -0,0 +1,189 @@
+import styles from '../styles/accountConnection'
+
+import React, { Component } from 'react'
+
+import AccountLoginForm from '../components/AccountLoginForm'
+import DataItem from '../components/DataItem'
+import ReactMarkdown from 'react-markdown'
+import Notifier from '../components/Notifier'
+import {popupCenter, waitForClosedPopup} from '../lib/popup'
+
+import { ACCOUNT_ERRORS } from '../lib/accounts'
+
+class AccountConnection extends Component {
+  constructor (props, context) {
+    super(props, context)
+    this.store = this.context.store
+  }
+
+  connectAccount ({login, password, folderPath}) {
+    const account = {
+      auth: {
+        login: login,
+        password: password
+      }
+    }
+
+    this.setState({ submitting: true })
+
+    return this.runConnection(account, folderPath)
+      .catch(error => this.handleError(error))
+  }
+
+  connectAccountOAuth (accountType) {
+    const cozyUrl =
+      `${window.location.protocol}//${document.querySelector('[role=application]').dataset.cozyDomain}`
+    const newTab = popupCenter(`${cozyUrl}/accounts/${accountType}/start`, `${accountType}_oauth`, 800, 800)
+    waitForClosedPopup(newTab, `${accountType}_oauth`)
+    .then(accountID => {
+      this.terminateOAuth(accountID)
+    })
+    .catch(error => {
+      this.setState({submitting: false, error: error.message})
+    })
+  }
+
+  terminateOAuth (accountID) {
+    const { t } = this.context
+    const { service } = this.state
+    const data = service.getData()
```

Comment:
Here we should be able to get the connector slug event when we are not in the service, if it's not the case it will break.